pkg/collector/corechecks/oracle-dbm/statements.go

```diff
@@ -607,29 +607,33 @@ func (c *Check) StatementMetrics() (int, error) {
 
 			oracleRows = append(oracleRows, oracleRow)
 
-			if _, found := c.fqtEmitted.Get(queryRow.QuerySignature); !found {
-				FQTDBMetadata := FQTDBMetadata{Tables: queryRow.Tables, Commands: queryRow.Commands}
-				FQTDB := FQTDB{Instance: c.cdbName, QuerySignature: queryRow.QuerySignature, Statement: SQLStatement, FQTDBMetadata: FQTDBMetadata}
-				FQTDBOracle := FQTDBOracle{
-					CDBName: c.cdbName,
-				}
-				FQTPayload := FQTPayload{
-					Timestamp:    float64(time.Now().UnixMilli()),
-					Host:         c.dbHostname,
-					AgentVersion: c.agentVersion,
-					Source:       common.IntegrationName,
-					Tags:         c.tagsString,
-					DBMType:      "fqt",
-					FQTDB:        FQTDB,
-					FQTDBOracle:  FQTDBOracle,
-				}
-				FQTPayloadBytes, err := json.Marshal(FQTPayload)
-				if err != nil {
-					log.Errorf("Error marshalling fqt payload: %s", err)
+			if c.fqtEmitted != nil {
+				if _, found := c.fqtEmitted.Get(queryRow.QuerySignature); !found {
+					FQTDBMetadata := FQTDBMetadata{Tables: queryRow.Tables, Commands: queryRow.Commands}
+					FQTDB := FQTDB{Instance: c.cdbName, QuerySignature: queryRow.QuerySignature, Statement: SQLStatement, FQTDBMetadata: FQTDBMetadata}
+					FQTDBOracle := FQTDBOracle{
+						CDBName: c.cdbName,
+					}
+					FQTPayload := FQTPayload{
+						Timestamp:    float64(time.Now().UnixMilli()),
+						Host:         c.dbHostname,
+						AgentVersion: c.agentVersion,
+						Source:       common.IntegrationName,
+						Tags:         c.tagsString,
+						DBMType:      "fqt",
+						FQTDB:        FQTDB,
+						FQTDBOracle:  FQTDBOracle,
+					}
+					FQTPayloadBytes, err := json.Marshal(FQTPayload)
+					if err != nil {
+						log.Errorf("Error marshalling fqt payload: %s", err)
+					}
+					log.Tracef("Query metrics fqt payload %s", string(FQTPayloadBytes))
+					sender.EventPlatformEvent(FQTPayloadBytes, "dbm-samples")
+					c.fqtEmitted.Set(queryRow.QuerySignature, "1", cache.DefaultExpiration)
 				}
-				log.Tracef("Query metrics fqt payload %s", string(FQTPayloadBytes))
-				sender.EventPlatformEvent(FQTPayloadBytes, "dbm-samples")
-				c.fqtEmitted.Set(queryRow.QuerySignature, "1", cache.DefaultExpiration)
+			} else {
+				log.Error("fqtEmitted = nil")
```

Comment:
When would `fqtEmitted` be nil? Would this be when the check is currently shutting down? If so, we might want to just include a note about that so it's more obvious to users/support how to interpret this error log.

Reply:
Yes the check was shutting down because the database restarted. I enhanced the error message.